src/operations/execute_operation.ts

```diff
@@ -286,6 +283,10 @@ async function tryOperation<
       }
       previousServer = server.description;
       previousOperationError = operationError;
+
+      // Reset timeouts
+      timeoutContext.serverSelectionTimeout?.clear();
+      timeoutContext.connectionCheckoutTimeout?.clear();
```

Comment:
clear timeouts to ensure correct setting of the `serverSelectionTimeoutMS` value